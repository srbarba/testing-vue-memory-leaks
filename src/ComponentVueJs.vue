<template>
  <div>Hello, {{ userName }}</div>
</template>

<script>
import Vue from 'vue'
import { mapGetters } from 'vuex'

export default Vue.extend({
  computed: {
    ...mapGetters(['userName']),
  },
})
</script>
