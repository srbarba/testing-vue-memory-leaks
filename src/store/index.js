export default {
  state: () => ({
    user: {
      name: undefined,
    },
  }),
  getters: {
    userName: (state) => state.user.name,
  },
  mutations: {
    setUser(state, user) {
      state.user = user
    },
  },
  actions: {},
}
