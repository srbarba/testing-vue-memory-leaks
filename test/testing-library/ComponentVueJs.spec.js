import { render } from '@testing-library/vue'
import ComponentVueJs from '../../src/ComponentVueJs.vue'
import store from '../../src/store'

describe('ComponentVueJs', () => {
  it('renders correctly', () => {
    const wrapper = render(ComponentVueJs, { store })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
