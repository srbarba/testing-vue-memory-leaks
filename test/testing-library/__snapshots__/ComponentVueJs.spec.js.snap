// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentVueJs renders correctly 1`] = `"<div>Hello, </div>"`;
